docs/tutorial/plugins.md

```diff
@@ -0,0 +1,547 @@
+---
+title: Create Plugins in JavaScript
+description: Plugins implemented in JavaScript are easily used across different projects, as they do not require any additional programs to be installed.
+---
+
+Jupyter Book uses the [MyST plugins infrastructure](xref:guide/plugins) to extend Jupyter Book's base functionality.
+Here are some things you can do with plugins:
+
+- Define a new directive or role to generate block-level or inline content.
+- Define a transform that modifies a page's content at build time.
+
+Plugins are natively written in JavaScript, or by writing an executable script in another language that follows [the MyST plugin protocol](xref:guide/executable-plugins).
+This tutorial shows you how to create a JavaScript plugin from scratch.
+
+## Create a basic Jupyter Book template
+
+We'll start by creating a basic template book that we can add to.
+
+🛠 First, create a new folder for a demo Jupyter Book (or use a pre-existing book).
+
+```shell
+$ mkdir plugin-demo
+$ cd plugin-demo
+```
+
+🛠 Next initialize a Jupyter Book in the folder to add a configuration file:
+
+```shell
+$ jupyter-book init
```

Comment:
In JB, we are currently leaning in to the `jupyter book` dispatch mechanism, so we should change this.